src/nvim/file_search.c

```diff
@@ -1396,7 +1396,12 @@ char *find_file_in_path_option(char *ptr, size_t len, int options, int first, ch
     // filename on the first call.
     if (first == true) {
       if (path_with_url(*file_to_find)) {
-        file_name = xstrdup(*file_to_find);
+        if (strncmp(*file_to_find, "file", 4) == 0) {
+          // convert "file://path/to/a" to "/path/to/a"
+          file_name = xstrdup(strchr(*file_to_find, '/'));
```

Comment:
the test treats `file://foo` as `foo`, but the comment here implies that it's treated as `/foo` ?